### PR TITLE
Fix issue where SpiralIn doesn't show elements.

### LINKED ISSUE #3514: `SpiralIn` is not working with the first Element
`SpiralIn` doesn't show the first Element sometimes, for examples just visit the current documentation of `SpiralIn`

- [ ] Tests are needed for SpiralIn to prevent this from happening again
- [ ] Implementation needs to be fixed

### PR DESCRIPTION
**Pull request by Gixtox**
## Motivation and Explanation: Why and how do your changes improve the library?

Fixes a bug where SpiralIn doesn't show elements that have a fill_opacity of 0. This also generates test data and adds a test to confirm the SpiralIn functionality. Resolve #3514

## Links to added or changed documentation pages

https://manimce--3589.org.readthedocs.build/en/3589/

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Review by JasonGrace2282 — DISMISSED** [review]
LGTM! Thanks for adding a test, too!

**Comment by JasonGrace2282** [comment]
Actually tested this with the following code and it doesn't seem to work?
```py
from manim import *


class Test(Scene):
    def construct(self):
        self.play(Create(Circle()))
        self.play(SpiralIn(Square().shift(RIGHT*3)))
```

https://github.com/ManimCommunity/manim/assets/110117391/493b8365-0a62-4dd3-9a4d-5f0d1c8aa83c

**Comment by Gixtox** [comment]
@JasonGrace2282 Thank you for taking a look at my pull request! On Github I'm able to see a video of a circle being drawn but not filled in, then a triangle and a square spiraling in from the sides. The triangle ends up on the left and the square ends up on the right. The video looks like I would expect given the code snippet. Can you please expand on what you're expecting?

**Comment by JasonGrace2282** [comment]
Oh I misread the PR fix the second time,  I thought it was fixing the abruptness of the animation.
Sorry for the confusion!

**Comment by Gixtox** [comment]
Sounds good! Thank you for such a fast response!

File: manim/animation/creation.py [hunk 1]
```diff
@@ -476,7 +476,7 @@ def interpolate_mobject(self, alpha: float) -> None:
         for shape in self.shapes:
             shape.restore()
             shape.save_state()
-            opacity = shape.get_fill_opacity()
+            opacity = shape.get_fill_opacity() or 1
```
Comment (behackl) [inline manim/animation/creation.py:479]: I'm not sure this is the correct solution here as it modifies the fill opacity of a given mobject -- this actually also happens in the test that you added; at the end of `SpiralIn`, the mobjects should be added to the scene with the opacity values they have been initialized with.

My suggestion for a slightly more complicated fix would be to split the current `opacity` into the typical `stroke_opacity` and `fill_opacity`, and then update both of them according to the rule given in `new_opacity`.

That way, no unexpected opacity changes happen.

**Review by behackl — CHANGES_REQUESTED** [review]
Hey there! Thanks for tackling this -- and kudos for identifying the problem! 👏 

I don't quite agree with the current solution and left a comment with a counter-proposal, could you please take a look?

File: manim/animation/creation.py (hunk 1 above)
Comment (Gixtox) [reply manim/animation/creation.py]: Howdy @behackl ! Thank you for the feedback. I like your idea much more. I wasn't aware the mobject attributes could be set separately like that. That __getattr__ function is nifty! 
I changed the fix so it lines up with your idea as I understand it, and I updated my test data to reflect that change.

One issue is it looks like my change did throw off the test_banner test. The frame comparison shows that the change causes the starting square to show on the top of the video. Do you think my fix is bad, is there something new to track down, or do we just need to regenerate the test data for test_banner?

**Comment by acremers22** [comment]
@behackl @brbisheng Following up.. we are still looking into this issue but are wondering if you have any additional details? To us, it seems that t_min and t_max are initialized in the ParametricFunction class in functions.py? Does this issue persist?

**Pull request merged**